### PR TITLE
Add `keep_firing_for` support to alerting rule

### PR DESCRIPTION
**Pull request by JHeilCoveo**
* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Notes

I didn't know if it was ok to take the work from another person that never did merge their PR so I wrote this PR from a clean state.
The support for `keep_firing_for` is mostly already present right now, this is more or less only UI stuff.

Fixes: #6165

## Changes

* Added `keep_firing_for` support to alerting rule

## Verification

Tested locally and updated e2e tests

### CODE REVIEW

File: pkg/rules/rulespb/rpc.proto [hunk 1]
```diff
@@ -119,13 +119,14 @@ message Alert {
...
     string query                              = 3 [(gogoproto.jsontag) = "query" ];
     double duration_seconds                   = 4 [(gogoproto.jsontag) = "duration" ];
-    ZLabelSet labels                          = 5 [(gogoproto.jsontag) = "labels", (gogoproto.nullable) = false ];
-    ZLabelSet annotations                     = 6 [(gogoproto.jsontag) = "annotations", (gogoproto.nullable) = false ];
-    repeated AlertInstance alerts             = 7 [(gogoproto.jsontag) = "alerts" ];
-    string health                             = 8 [(gogoproto.jsontag) = "health" ];
-    string last_error                         = 9 [(gogoproto.jsontag) = "lastError,omitempty" ];
-    double evaluation_duration_seconds        = 10 [(gogoproto.jsontag) = "evaluationTime" ];
-    google.protobuf.Timestamp last_evaluation = 11 [(gogoproto.jsontag) = "lastEvaluation", (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+    double keep_firing_for_seconds            = 5 [(gogoproto.jsontag) = "keepFiringFor" ];
```
Comment (JHeilCoveo) [inline pkg/rules/rulespb/rpc.proto:122]: I never did code anything with protocol buffer so I don't know if it's frowned upon to add a new attribute in the middle of an already existing structure.
I can change this if it causes any issue.

Comment (JHeilCoveo) [reply pkg/rules/rulespb/rpc.proto]: Got my anser, it broke `TestQueryCompatibilityWithPreInfoAPI` e2e test!

**Comment by JHeilCoveo** [comment]
Could the e2e be flaky or did my change break something seemingly unrelated ?

**Comment by yeya24** [comment]
It is a flaky one. I just retried the test

**Review by yeya24 — APPROVED** [review]
Thanks! LGTM

**Pull request merged**